packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -945,6 +902,11 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             }
         }
 
+        if (this.hostPolicy.summarizerClient) {
```

Comment:
Same code is in getSnapshotTree() Should instead we have one instance of this code?
I think the reason code structured the way it is is due to early return:
```
        if (commits && commits[".app"]) {
            // The latest snapshot is a summary
            // attempt to read .protocol from commits for backwards compat
            return this.readSummaryTree(tree.id, commits[".protocol"] || hierarchicalTree.trees[".protocol"], commits[".app"] as string);
        }
```
If we had only one return path in getSnapshotTree() then we would not need to have that statement here.
It would be great to make such change and have logic in one place

Reply:
done